manifests/opsmanager.pp

```diff
@@ -1,17 +1,38 @@
-# This installs Ops Manager. See README.md for more info.
-
+# This installs Ops Manager
+#
+# @param user
+#   The user that owns the config file
+#
+# @param group
+#   The group that owns the config file
+#
+# @param opsmanager_url
+#   The fully qualified url where opsmanager runs. Must include the port.
+#
+# @param opsmanager_mongo_uri
+#   Full URI where the Ops Manager application mongodb server(s) can be found.
+#
+# @param ca_file
+#   Ca file for secure connection to backup agents.
+#
+# @param pem_key_file
+#   Pem key file containing the cert and private key used for secure connections to backup agents.
+#
+# @param pem_key_password
+#   The password to the pem key file.
+#
 class mongodb::opsmanager (
-  String[1] $user                                = $mongodb::opsmanager::params::user,
-  String[1] $group                               = $mongodb::opsmanager::params::group,
-  Enum['running', 'stopped'] $ensure             = $mongodb::opsmanager::params::ensure,
-  String[1] $package_name                        = $mongodb::opsmanager::params::package_name,
-  String[1] $package_ensure                      = $mongodb::opsmanager::params::package_ensure,
-  Boolean $service_enable                        = $mongodb::opsmanager::params::service_enable,
-  Boolean $service_manage                        = $mongodb::opsmanager::params::service_manage,
-  String[1] $service_name                        = $mongodb::opsmanager::params::service_name,
-  Stdlib::Httpurl $download_url                  = $mongodb::opsmanager::params::download_url,
-  String[1] $mongo_uri                           = $mongodb::opsmanager::params::mongo_uri,
-  Stdlib::Httpurl $opsmanager_url                = $mongodb::opsmanager::params::opsmanager_url,
+  String[1] $user                                = 'mongodb-mms',
+  String[1] $group                               = 'mongodb-mms',
+  Enum['running', 'stopped'] $ensure             = 'running',
+  String[1] $package_name                        = 'mongodb-mms',
+  String[1] $package_ensure                      = 'present',
+  Boolean $service_enable                        = true,
+  Boolean $service_manage                        = true,
+  String[1] $service_name                        = 'mongodb-mms',
+  Stdlib::Httpurl $download_url                  = 'https://downloads.mongodb.com/on-prem-mms/rpm/mongodb-mms-4.0.1.50101.20180801T1117Z-1.x86_64.rpm',
```

Comment:
This does make me suspect we need to do some deb/rpm handling in params

Reply:
Are you leaving this for later.

Reply:
Yes. It was already like this before the refactor so I don't want to add more features in this PR. However, we can generate a more readable REFERENCE.md which should make it easy for Debian admins to see it will probably not work for them so I already see it as an improvement.